### DISCUSSION TITLE
Why use flatpak theme rather than deb package

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [SpiralLinux/SpiralLinux-project#302] [Q&A]
I read such lines in hooks:
```
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
flatpak install -y --noninteractive --system runtime/org.kde.KStyle.Kvantum/x86_64/5.15-21.08
flatpak install -y --noninteractive --system org.gtk.Gtk3theme.Numix
flatpak override --env=QT_STYLE_OVERRIDE=kvantum
flatpak override --filesystem=xdg-config/Kvantum:ro
```
This is to install theme by flatpak. Themes can also be installed by apt, so why use flatpak instead?

Besides, what two lines of `flatpak override` mean to do?

**Comment by geckolinux** [comment]
Hi @CicadaSeventeen , the themes installed via Flatpak are only for theming apps installed via Flatpak. SpiralLinux also includes other packages preinstalled with APT for theming the apps from the Debian repos.

In the case of GTK themes, since Flatpak apps are sandboxed they can not use APT installed themes in `/usr/share/themes/` and therefore require internal Flatpak theme packages. The `flatpak override --filesystem=` command will open the sandbox to allow Flatpak apps to access that directory in the filesystem, but it has many strictly prohibited locations, including `/usr/share/themes/` , so there is no way to share the APT installed GTK themes with Flatpak.

In the case of Qt apps, SpiralLinux uses an environment variable in `/etc/environment` that makes normal APT installed QT4/5 apps imitate the appearance of the active GTK2 theme. But that doesn't work in the Flatpak environment, which requires explicitly installing Qt theme packages from Flatpak and configuring environment variables to set the theme for Qt Flatpak apps. There are no available Qt themes that look similar to the Numix GTK theme that SpiralLinux uses, so I use Kvantum, which is a special Qt theme engine that has its own internal themes and allows the user to choose widget style and colors. Its config files are located in the user's `~/.config/Kvantum/` directory. So for the Flatpak environment I set the Qt theme to Kvantum and then include `~/.config/Kvantum/` by default for all users (via `/etc/skel/`) pre-configured with a theme that blends fairly well with Numix, and I allow Flatpak to access that from its sandbox with `flatpak override --filesystem=xdg-config/Kvantum:ro` .

Basically, Linux theming is a confusing mess, and it's even more confusing and messy in Flatpak. ;-) This might help too:
https://www.reddit.com/r/flatpak/comments/y9jmqj/the_general_flatpak_qt_and_gtk_theming_guide/
